.github/workflows/nightly.yaml

```diff
@@ -28,61 +28,26 @@ jobs:
         name: Checkout ${{ matrix.branch }} branch
         run: git switch ${{ matrix.branch }}
 
-      # check changes in this commit for regex include and exclude matches; pipe to an env var
-      - name: Check for changes to build
-        run: |
-          # don't fail if nothing returned by grep
-          set +e 
-          CHANGES="$(git diff --name-only HEAD~1 | \
-            grep -E "workflows/.+-container-build.yaml|Makefile|bundle/|config/|go.mod|go.sum|.+\.go|docker/|\.dockerignore" | \
-            grep -v -E ".+_test.go|/.rhdh/")";
-          echo "Changed files for this commit:"
-          echo "=============================="
-          echo "$CHANGES"
-          echo "=============================="
-          {
-            echo 'CHANGES<<EOF'
-            echo $CHANGES
-            echo EOF
-          } >> "$GITHUB_ENV"
-
       - name: Setup Go
-        # run this stage only if there are changes that match the includes and not the excludes
-        if: ${{ env.CHANGES != '' }}
         uses: actions/setup-go@0c52d547c9bc32b1aa3301fd7a9cb496313a4491 # v5
         with:
           go-version-file: 'go.mod'
 
       - name: Determine built operator image
-        # run this stage only if there are changes that match the includes and not the excludes
-        if: ${{ env.CHANGES != '' }}
-        run: |
-          SHORT_SHA=$(git rev-parse --short HEAD)
-          BASE_VERSION=$(grep -E "^VERSION \?=" Makefile | sed -r -e "s/.+= //") # 0.1.0
-          echo "OPERATOR_IMAGE=quay.io/janus-idp/operator:${BASE_VERSION}-${SHORT_SHA}" >> $GITHUB_ENV
-
-      - name: Wait until image exists in registry or timeout is reached
-        # run this stage only if there are changes that match the includes and not the excludes
-        if: ${{ env.CHANGES != '' }}
-        timeout-minutes: 10
         run: |
-          echo "Waiting until operator image is found or timeout expires: ${{ env.OPERATOR_IMAGE }}..."
-          until ${CONTAINER_ENGINE} image pull "${{ env.OPERATOR_IMAGE }}"; do
-            sleep 2
-            echo ...
-          done
-          echo "... operator image found: ${{ env.OPERATOR_IMAGE }}."
+          latestNext="next"
+          # for main branch, use next tags; for 1.x branches, use :latest tags
+          if [[ $(git rev-parse --abbrev-ref HEAD) != "main" ]]; then
+            latestNext="latest" 
+          fi
+          echo "OPERATOR_IMAGE=quay.io/janus-idp/operator:${latestNext}" >> $GITHUB_ENV
```

Comment:
so this workflow doesn't build anything, it just runs tests on the last latest (or next) tagged image? 

why not couple this with the action that creates the new container image (and `:latest` or `:next` tag) ? then you'd get results immediately after each build.

Or is the goal to collect several CI builds into a single daily E2E check, to save resources? If so, +1.

Reply:
Yup, the goal is indeed to collect several commits and run it on a nightly basis to save resources.